core/app/c/[communitySlug]/pubs/create/page.tsx

```diff
@@ -0,0 +1,95 @@
+import type { Metadata } from "next";
+
+import { notFound } from "next/navigation";
+
+import type { CommunitiesId, PubsId, StagesId } from "db/public";
+import { Capabilities } from "db/src/public/Capabilities";
+import { MembershipType } from "db/src/public/MembershipType";
+import { Button } from "ui/button";
+
+import { ContentLayout } from "~/app/c/[communitySlug]/ContentLayout";
+import { PageTitleWithStatus } from "~/app/components/pubs/PubEditor/PageTitleWithStatus";
+import { PubEditor } from "~/app/components/pubs/PubEditor/PubEditor";
+import { getPageLoginData } from "~/lib/authentication/loginData";
+import { userCan } from "~/lib/authorization/capabilities";
+import { getCommunityBySlug } from "~/lib/db/queries";
+
+export async function generateMetadata({
+	params: { communitySlug },
+}: {
+	params: { pubId: string; communitySlug: string };
+}): Promise<Metadata> {
+	const community = await getCommunityBySlug(communitySlug);
+	if (!community) {
+		return { title: "Community Not Found" };
+	}
+
+	return { title: `Create pub in ${community.name}` };
+}
+
+export default async function Page({
+	params,
+	searchParams,
+}: {
+	params: { communitySlug: string };
+	searchParams: Record<string, string>;
+}) {
+	const { communitySlug } = params;
+
+	const { user } = await getPageLoginData();
+
+	const community = await getCommunityBySlug(communitySlug);
+
+	if (community === null) {
+		notFound();
+	}
+
+	const communityId = community.id as CommunitiesId;
+
+	const canCreatePub = await userCan(
+		Capabilities.createPub,
+		{
+			type: MembershipType.community,
+			communityId,
+		},
+		user.id
+	);
+
+	if (!canCreatePub) {
+		return null;
```

Comment:
not sure if we have standardized behavior for what things should look like if `!userCan`?

Reply:
I think calling `notFound()` or [explicitly redirecting them to our unauthorized page](https://github.com/pubpub/platform/blob/main/core/app/c/%5BcommunitySlug%5D/fields/page.tsx#L57) is what we've been doing

When we upgrade to next 15 I would like to use https://nextjs.org/docs/app/api-reference/functions/forbidden or https://nextjs.org/docs/app/api-reference/functions/unauthorized